services/graphite/service.go

```diff
@@ -223,8 +265,10 @@ func (s *Service) openTCPServer() (net.Addr, error) {
 // handleTCPConnection services an individual TCP connection for the Graphite input.
 func (s *Service) handleTCPConnection(conn net.Conn) {
 	defer conn.Close()
+	defer removeConnection(conn)
 	defer s.wg.Done()
```

Comment:
Makes sense.

On Thursday, September 3, 2015, dgnorton <notifications@github.com> wrote:

> In services/graphite/service.go
> <https://github.com/influxdb/influxdb/pull/3969#discussion_r38712841>:
>
> > @@ -223,8 +265,10 @@ func (s *Service) openTCPServer() (net.Addr, error) {
> >  // handleTCPConnection services an individual TCP connection for the Graphite input.
> >  func (s *Service) handleTCPConnection(conn net.Conn) {
> >  	defer conn.Close()
> > +	defer removeConnection(conn)
> >  	defer s.wg.Done()
>
> I know this wasn't part of this PR but should s.wg.Done() be moved up so
> it's the first defer since defered funcs execute in LIFO order?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdb/influxdb/pull/3969/files#r38712841>.
>